pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -410,3 +374,120 @@ func executeAgentCommands(commands []AgentCommand, con cli.Connection, agentStor
 		}
 	}
 }
+
+func executeApiOperations(commands []AgentCommand, con cli.Connection, agentStore *AsyncAgentStore, domainName string) {
+	for _, command := range commands {
```

Comment:
This loop is not needed. All the different commands call the same libvirt API `domain.GetGuestInfo()`. We can call it once and get all the results:
```
flags := libvirt.DOMAIN_GUEST_INFO_INTERFACES |
	libvirt.DOMAIN_GUEST_INFO_OS |
	libvirt.DOMAIN_GUEST_INFO_HOSTNAME |
	libvirt.DOMAIN_GUEST_INFO_TIMEZONE |
	libvirt.DOMAIN_GUEST_INFO_USERS
guestInfo, err := domain.GetGuestInfo(flags, 0)
```

Reply:
I think it should be better this way. Thanks!